app/src/main/java/com/kamron/pogoiv/ScreenGrabber.java

```diff
@@ -144,4 +114,55 @@ private Bitmap getBitmap(ByteBuffer buffer, int pixelStride, int rowPadding) {
 
         return bmp;
     }
+
+    public @Nullable @ColorInt int[] grabPixels(Point[] points) {
+        Image image = null;
+        try {
+            //Note: imageReader shouldn't be null, but apparently sometimes is.
+            //Let's allow this to still happen.
```

Comment:
You have, but you said you want to change the timer we use *after* the big refactoring.